File: apps/web/src/lib/api/tts.ts
import { apiClient, type ApiClient, type ApiRequestOptions } from './client'
import type { ApiResponse } from './types'

export interface TtsVoice {
  id: string
  name: string
  language: string
  provider: string
  model: string
  available: boolean
}

export interface TtsCapabilities {
  voices: TtsVoice[]
  languages: string[]
}

export const getTtsCapabilities = async (
  options?: ApiRequestOptions,
  client: ApiClient = apiClient,
): Promise<TtsCapabilities> =>
  (await client.get<ApiResponse<TtsCapabilities>>('/api/v1/tts/capabilities', options)).data

File: apps/api/app/schemas/tts.py
from app.schemas.common import ApiModel


class TtsVoiceView(ApiModel):
    id: str
    name: str
    language: str
    provider: str
    model: str
    available: bool


class TtsCapabilitiesView(ApiModel):
    voices: list[TtsVoiceView]
    languages: list[str]
